packages/opentelemetry-api/src/api/propagation.ts

```diff
@@ -79,6 +95,19 @@ export class PropagationAPI {
     getter: GetterFunction<Carrier> = defaultGetter,
     context = contextApi.active()
   ): Context {
-    return this._propagator.extract(context, carrier, getter);
+    return this._getGlobalPropagator().extract(context, carrier, getter);
+  }
+
+  /** Remove the global propagator */
+  public disable() {
+    delete _global[GLOBAL_PROPAGATION_API_KEY];
+  }
+
+  private _getGlobalPropagator(): HttpTextPropagator {
+    if (!_global[GLOBAL_PROPAGATION_API_KEY]) {
+      return NOOP_HTTP_TEXT_PROPAGATOR;
+    }
+
+    return _global[GLOBAL_PROPAGATION_API_KEY]!(API_VERSION);
```

Comment:
`_global[GLOBAL_PROPAGATION_API_KEY]!(API_VERSION) || NOOP_HTTP_TEXT_PROPAGATOR` ?

Reply:
We need to guard against the case where the global is not yet set.